src/test/java/com/a2/backend/controller/UserControllerTest.java

```diff
@@ -401,4 +401,60 @@ void Test010_GivenAValidTokenAndUserWhenConfirmingUserThenStatusOKisReturned() {
         assertEquals(
                 "There is an existing user with the nickname " + anotherNickname, errorMessage);
     }
+
+    @Test
+    void GivenAvalidUserWhenWantToRecoverPasswordThenReturnStatusOk() {
+        UserCreateDTO userCreateDTO =
+                UserCreateDTO.builder()
+                        .nickname(nickname)
+                        .email(email)
+                        .biography(biography)
+                        .password(password)
+                        .confirmationToken(confirmationToken)
+                        .build();
+        PasswordRecoveryDTO passwordRecoveryDTO =
+                PasswordRecoveryDTO.builder()
+                        .passwordRecoveryToken("")
+                        .newPassword("NewPassword001")
+                        .build();
+        String validConfirmationToken = "token001";
+
+        HttpEntity<UserCreateDTO> request = new HttpEntity<>(userCreateDTO);
+        HttpEntity<PasswordRecoveryDTO> requestPasswordRecovery =
+                new HttpEntity<>(passwordRecoveryDTO);
+
+        val postResponse = restTemplate.exchange(baseUrl, HttpMethod.POST, request, User.class);
+        assertEquals(HttpStatus.CREATED, postResponse.getStatusCode());
+
+        User userToBeUpdated = postResponse.getBody();
+
+        val getResponse =
+                restTemplate.exchange(
+                        String.format(
+                                "%s/%s/%s/%s",
+                                baseUrl,
+                                confirmationUrl,
+                                userToBeUpdated.getId(),
+                                validConfirmationToken),
+                        HttpMethod.GET,
+                        null,
+                        User.class);
+        assertEquals(HttpStatus.OK, getResponse.getStatusCode());
+        passwordRecoveryDTO.setPasswordRecoveryToken(userToBeUpdated.getPasswordRecoveryToken());
+
+        User activatedUserToBeUpdated = getResponse.getBody();
+        System.out.println(userToBeUpdated.getPassword());
```

Comment:
Why sout?

Reply:
I forgot to delete that, I was getting null and wanted to check if it was the password's problem or what was it. Found the problem and fixed it, but forgot to delete the sout